andela/devrepublic-bn-backend
# #170947549 reset password feature

#### What does this PR do?
Adds the user password reset functionality
#### Description of Task to be completed?
- add the forgot password endpoint
- add the reset password endpoint
- configured SendGrid to send email
- add the verify token function
#### How should this be manually tested?
- clone this repo
- run npm install
- set up the database(follow the readme)
-  use the forgot password route and follow the instruction
- check your email 
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#170947549
#### Screenshots (if appropriate)
<img width="646" alt="Screen Shot 2020-02-18 at 20 31 38" src="https://user-images.githubusercontent.com/51261911/74766385-e9869880-528d-11ea-8659-9bc3d718b173.png">
<img width="664" alt="Screen Shot 2020-02-18 at 20 32 56" src="https://user-images.githubusercontent.com/51261911/74766555-fc00d200-528d-11ea-8cdb-32cd790efbb4.png">
<img width="1111" alt="Screen Shot 2020-02-18 at 20 42 44" src="https://user-images.githubusercontent.com/51261911/74767611-d83e8b80-528f-11ea-86ef-ea15192a14f7.png">
<img width="1073" alt="Screen Shot 2020-02-18 at 20 45 08" src="https://user-images.githubusercontent.com/51261911/74767663-f5735a00-528f-11ea-96c0-03a76c4f5336.png">
<img width="1131" alt="Screen Shot 2020-02-18 at 20 44 37" src="https://user-images.githubusercontent.com/51261911/74767719-0f14a180-5290-11ea-833b-99101fc5f28f.png">
#### Questions:
